sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -30,7 +30,7 @@ public ClientDiagnostics(ClientOptions options)
         }
 
         public ClientDiagnostics(string optionsNamespace, string? providerNamespace, DiagnosticsOptions diagnosticsOptions)
-            : base(optionsNamespace, providerNamespace, diagnosticsOptions.IsDistributedTracingEnabled)
+            : base(optionsNamespace, providerNamespace, diagnosticsOptions.IsDistributedTracingEnabled, false)
```

Comment:
Should this respect the value in diagnosticOptions?

Reply:
we can't until we ship new Azure.Core:

> E.g. Azure.Identity uses shared ClientDiagnostics source file. Identity has package reference to Azure.Core, i.e. it uses released version of Core.
> So when I define new property in DiagnositcOptions, Identity can't use it yet, i.e. ClientDiagnostics (being part of the identity package) does not know about it too.

Reply:
Maybe I'm misunderstanding, but with shared source the file will be compiled into each library that it is shared with, so dependencies don't come into play.

Reply:
Ah nevermind - I see your point. The public property is in DiagnosticOptions which is not yet shipped.

Reply:
Usually we would update to a project reference in this case or file an issue so that we don't forget to update this once Core ships.